Signal/src/Models/TSMessageAdapaters/TSVideoAttachmentAdapter.m

```diff
@@ -81,59 +84,97 @@ - (BOOL)isVideo {
+{
+    OWSAssert([NSThread isMainThread]);
+
+    self.audioProgressSeconds = progress;
+    self.audioDurationSeconds = duration;
+
+    [self updateAudioBottomLabel];
 }
 
-- (void)setAudioProgressFromFloat:(float)progress {
-    dispatch_async(dispatch_get_main_queue(), ^{
-      if (!isnan(progress)) {
-          [_waveform setProgress:progress];
-          [_waveform generateWaveforms];
-          [_waveform setNeedsDisplay];
-      }
-    });
+- (NSString *)formatTime:(long)timeSeconds
+{
+    long seconds = timeSeconds % 60;
+    long minutes = (timeSeconds / 60) % 60;
+    long hours = timeSeconds / 3600;
+
+    if (hours > 0) {
+        return [NSString stringWithFormat:@"%ld:%02ld:%02ld", hours, minutes, seconds];
+    } else {
+        return [NSString stringWithFormat:@"%ld:%02ld", minutes, seconds];
+    }
+}
+
+- (void)updateAudioBottomLabel
+{
+    if (self.isAudioPlaying && self.audioProgressSeconds > 0 && self.audioDurationSeconds > 0) {
+        self.audioBottomLabel.text = [NSString stringWithFormat:@"%@ / %@",
+                                               [self formatTime:(long)round(self.audioProgressSeconds)],
+                                               [self formatTime:(long)round(self.audioDurationSeconds)]];
+    } else {
+        NSError *error;
+        unsigned long long fileSize =
+            [[NSFileManager defaultManager] attributesOfItemAtPath:self.attachment.filePath error:&error].fileSize;
+        OWSAssert(!error);
+        NSString *bottomText = [ViewControllerUtils formatFileSize:fileSize];
+        self.audioBottomLabel.text = bottomText;
+    }
+}
+
+- (void)setAudioIcon:(UIImage *)image
+{
+    [_audioPlayPauseButton setImage:image forState:UIControlStateNormal];
+    [_audioPlayPauseButton setImage:image forState:UIControlStateDisabled];
+    _audioPlayPauseButton.layer.opacity = 0.8f;
 }
 
 - (void)setAudioIconToPlay {
-    [_audioPlayPauseButton
-        setBackgroundImage:[UIImage imageNamed:(_incoming ? @"audio_play_button_blue" : @"audio_play_button")]
-                  forState:UIControlStateNormal];
+    [self setAudioIcon:[UIImage imageNamed:(self.incoming ? @"audio_play_black_40" : @"audio_play_white_40")]];
 }
 
 - (void)setAudioIconToPause {
-    [_audioPlayPauseButton
-        setBackgroundImage:[UIImage imageNamed:(_incoming ? @"audio_pause_button_blue" : @"audio_pause_button")]
-                  forState:UIControlStateNormal];
+    [self setAudioIcon:[UIImage imageNamed:(self.incoming ? @"audio_pause_black_40" : @"audio_pause_white_40")]];
+}
+
+#pragma mark - JSQMessageMediaData protocol
+
+- (CGFloat)bubbleHeight
+{
+    return 35.f;
 }
 
-- (void)removeDurationLabel {
-    [_durationLabel removeFromSuperview];
+- (CGFloat)iconSize
+{
+    return 40.f;
 }
 
-#pragma mark - JSQMessageMediaData protocol
+- (CGFloat)vMargin
+{
+    return 10.f;
+}
+
+- (UIColor *)audioTextColor
+{
+    return (self.incoming ? [UIColor colorWithWhite:0.2 alpha:1.f] : [UIColor whiteColor]);
```

Comment:
See my other comment but I don't intend to use this color elsewhere. The attachment bubbles just seemed way too visually overwhelming with "all of that black ink" and three separate elements crammed into a single bubble.  Ideally we can revisit the colors, type, bubble shapes and size, etc. when we replace JSQ.